File: DSRFQ.Web/Modules/Material/RawMaterialCosts/RawMaterialCostsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Material.Columns;

[ColumnsScript("Material.RawMaterialCosts")]
[BasedOnRow(typeof(RawMaterialCostsRow), CheckNames = true)]
public class RawMaterialCostsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string MaterialCode { get; set; }
    public int MaterialTemperId { get; set; }
    public string WeightDimensionUnitCode { get; set; }
    public string CurrencyCode { get; set; }
    public decimal UnitPrice { get; set; }
    public DateTime FromDate { get; set; }
    public DateTime ToDate { get; set; }
  
    [DateTimeFormatter(DisplayFormat = "yyyy-MM-dd HH:mm"), Width(120)]
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    [DateTimeFormatter(DisplayFormat = "yyyy-MM-dd HH:mm"), Width(120)]
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    [DateTimeFormatter(DisplayFormat = "yyyy-MM-dd HH:mm"), Width(120)]
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}

File: DSRFQ.Web/Modules/ServerTypes/Material/RawMaterialCostsColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { RawMaterialCostsRow } from './RawMaterialCostsRow';

export interface RawMaterialCostsColumns {
    Id: Column<RawMaterialCostsRow>;
    MaterialId: Column<RawMaterialCostsRow>;
    MaterialTemperId: Column<RawMaterialCostsRow>;
    MaterialShapeId: Column<RawMaterialCostsRow>;
    WeightDimensionUnitId: Column<RawMaterialCostsRow>;
    CurrencyId: Column<RawMaterialCostsRow>;
    UnitPrice: Column<RawMaterialCostsRow>;
    FromDate: Column<RawMaterialCostsRow>;
    ToDate: Column<RawMaterialCostsRow>;
    CompanyId: Column<RawMaterialCostsRow>;
    GroupId: Column<RawMaterialCostsRow>;
    InsertDate: Column<RawMaterialCostsRow>;
    InsertUserId: Column<RawMaterialCostsRow>;
    UpdateDate: Column<RawMaterialCostsRow>;
    UpdateUserId: Column<RawMaterialCostsRow>;
    DeleteDate: Column<RawMaterialCostsRow>;
    DeleteUserId: Column<RawMaterialCostsRow>;
    IsActive: Column<RawMaterialCostsRow>;
}

export class RawMaterialCostsColumns extends ColumnsBase<RawMaterialCostsRow> {
    static readonly columnsKey = 'Material.RawMaterialCosts';
    static readonly Fields = fieldsProxy<RawMaterialCostsColumns>();
}